lib/spack/spack/test/build_systems.py

```diff
@@ -320,6 +320,13 @@ def test_cmake_std_args_hip(self, default_mock_concretization):
         option = spack.build_systems.cmake.CMakeBuilder.define_hip_architectures(s.package)
         assert "-DCMAKE_HIP_ARCHITECTURES:STRING=gfx900" == option
 
+    def test_cmake_dependent_args(self, concretize_and_setup, monkeypatch):
+        def cmake_arg_check(*options):
+            assert "-DDEP_TEST_ARG=ON" in options
+
+        s = concretize_and_setup("cmake-client-dependent")
+        monkeypatch.setattr(s.package.module, "cmake", cmake_arg_check)
```

Comment:
My suggestion makes this test a bit more complex than the others in this module:

1. Right now, `cmake_arg_check` isn't being run
2. I asked you to verify things by intercepting at the utility level (i.e. the executables that Spack invokes), but that is different than the other checks here
3. I think [2] is more robust, but to actually perform it, you would have to structure this test more like a test in the `install.py` test module (i.e. actually install the package) - AFAIK we don't have anything for sandboxing the execution of individual phases in a test environment
4. In particular I think [2] is useful because I wanted to check what would happen if I overrode `cmake` in `test-cmake-dependent-args` (I was assuming it would fail, but I wanted to make sure of that and also check a couple other things to write out in the commit message on how users should access `dependent_args` if they need to override the `cmake` phase for some reason (i.e. `PackageClass.cmake()`, not `the_package_module.cmake()`)